### DISCUSSION TITLE
My new work: Control for Transmission

### DISCUSSION THREAD

**Discussion by hanguokai** [transmission/transmission#4377] [Show and tell]
<img width="883" alt="webui-en-raw" src="https://user-images.githubusercontent.com/776467/208041240-a470dea1-98c3-494e-ac36-56fcbbaade34.png">

### Overview
A concise, lightweight and high-performance Transmission Manager and Web Extension. It supports managing multiple remote or local Transmission at the same time, as well as adding magnet links from BitTorrent websites.

### Features
1. Concise, lightweight and high-performance: Removed some not commonly used functions; The installation package size is small; Fast page loading, low CPU and memory usage.
2. Support simultaneous management of download tasks in multiple remote Transmissions.
3. Support adding download tasks by right-clicking on magnet links in Web pages.
4. This extension doesn't require "Read and change all your data on all websites" permission.

### Install Links
Chrome: https://chrome.google.com/webstore/detail/hfhhlbapamidaibjcgdmeidlcoaoadme
Edge: https://microsoftedge.microsoft.com/addons/detail/pefbobclhiokohhohjobgiaelbdcnaeb

**Comment by ckerr** [comment]
Looks great! Is there a repo for the source code?

**Reply by hanguokai** [reply]
Hi, this is a new work I wrote in the last month, and it has not been open sourced yet. If Transmission community is willing to accept this project, I can open source it to https://github.com/transmission/a-sub-project . This is a pure frontend tool (no backend), so no matter whether it is open source or not, there is no secret.

**Comment by GaryElshaw** [comment]
The icon is very little. Here is one i made that didn't get used for the project that might serve better? Because of GitHub limitations, you'll have to remove .txt from the file.

![SCR-20221216-s2e](https://user-images.githubusercontent.com/69029666/208043600-9c15a8e5-c5cb-4eb3-8910-c3a00eaf65bb.png)

[Transmission-icon.svg.txt](https://github.com/transmission/transmission/files/10243455/Transmission-icon.svg.txt)

**Reply by hanguokai** [reply]
Thanks. Unfortunately, chrome extension can't support svg file as extension icon at the moment. The icon size on the browser toolbar is 32x32(16x16), so complex icon is difficult to display clearly.

**Reply by GaryElshaw** [reply]
32x32 png

![Transmission-icon](https://user-images.githubusercontent.com/69029666/208047184-20c633ea-0a8a-4499-aa4c-f4f71491433e.png)

**Comment by hanguokai** [comment]
I noticed that RPC api doesn't response [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) http header. So I use extension host permission to overcome this. If PRC api support CORS, most functions(except browser context menu function) can be implemented as a pure Web app (PWA).
